### PR TITLE
Edge cases

### PR DESCRIPTION
**Pull request by BartlomiejMilewski**

### CODE REVIEW

File: src/test/java/com/example/Bloggy/BlogPostControllerTest.java [hunk 1]
```diff
@@ -64,6 +66,19 @@ void testGetBlogPostById() throws Exception {
     }
 
 
+    @Test
+    @DisplayName("GET /posts/1 - Not found")
+    void testGetBlogPostByIdNotFound() throws Exception {
```
Comment (TomaszLaski) [inline src/test/java/com/example/Bloggy/BlogPostControllerTest.java]: Imo better name would be: GetBlogPostByIdShouldReturnNotFoundIfNoPostFound() test names should best reflect their functions, in complex applications tests are counted in thousands, we want to easily determine what does not work simply by looking on name of failed test

File: src/test/java/com/example/Bloggy/BlogPostControllerTest.java [hunk 2]
```diff
@@ -116,6 +131,43 @@ void testUpdateBlogPost() throws Exception {
 
     }
 
+    @Test
+    @DisplayName("PUT /posts/{id} - Can not update, post not found")
+    void testUpdateBlogPostIfPostNotFound() throws Exception {
```
Comment (TomaszLaski) [inline src/test/java/com/example/Bloggy/BlogPostControllerTest.java]: +1

File: src/test/java/com/example/Bloggy/BlogPostControllerTest.java [hunk 3]
```diff
@@ -116,6 +131,43 @@ void testUpdateBlogPost() throws Exception {
...
+//
+//        doReturn(null).when(service).findPostById(1);
+//        doReturn(null).when(service).updatePost(any(BlogPost.class));
+//
+//        mockMvc.perform(put("/bloggy/posts/{id}", 1)
+//                        .contentType(MediaType.APPLICATION_JSON)
+//                        .content("{\"author\":\"Test Author\",\"tags\":\"Test tags\",\"title\":\"Test title\",\"content\":\"Test content\"}"))
+//
+//                .andExpect(status().isConflict());
+//    }
```
Comment (TomaszLaski) [inline src/test/java/com/example/Bloggy/BlogPostControllerTest.java]: remove comments from code

File: src/test/java/com/example/Bloggy/BlogPostControllerTest.java [hunk 4]
```diff
@@ -132,6 +184,16 @@ void testDeleteBlogPost() throws Exception {
 
     }
 
+    @Test
+    @DisplayName("DELETE /posts/{id} - Not Found")
+    void testDeleteBlogPostNotFound() throws Exception {
```
Comment (TomaszLaski) [inline src/test/java/com/example/Bloggy/BlogPostControllerTest.java]: +1

File: src/main/java/com/example/Bloggy/controller/BlogPostController.java [hunk 5]
```diff
@@ -37,26 +37,45 @@ public ResponseEntity<BlogPost> addPost(@RequestBody BlogPost blogPost) {
...
     public ResponseEntity<BlogPost> updatePost(@PathVariable Integer id, @RequestBody BlogPost blogPostDetails) {
         BlogPost blogPost = postService.findPostById(id);
 
-        blogPost.setAuthor(blogPostDetails.getAuthor());
-        blogPost.setTags(blogPostDetails.getTags());
-        blogPost.setTitle(blogPostDetails.getTitle());
-        blogPost.setContent(blogPostDetails.getContent());
+        if (blogPost == null) {
+            return ResponseEntity.notFound().build();
+//        } else if (blogPost != postService.findPostById(1)) {
```
Comment (TomaszLaski) [inline src/main/java/com/example/Bloggy/controller/BlogPostController.java]: please remove commnets from code

**Review by TomaszLaski — CHANGES_REQUESTED** [review]
great job, just change the names of the tests to reflect their functionality and comments in the code.

**Review by TomaszLaski — APPROVED** [review]
